commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/CommandAPIBukkitConfig.java

```diff
@@ -50,6 +51,19 @@ public CommandAPIBukkitConfig useMojangMappings(boolean useMojangMappings) {
 		return this;
 	}
 
+	/**
+	 * Sets whether the CommandAPI should use the plugin's name as the default namespace namespace
+	 * <p>
+	 * If called, any call to {@link CommandAPIConfig#setNamespace(String)} will be ignored
+	 *
+	 * @return this CommandAPIBukkitConfig
+	 */
+	public CommandAPIBukkitConfig usePluginNamespace() {
+		super.setNamespace(plugin.getName().toLowerCase());
+		super.usePluginNamespace = true;
```

Comment:
I was thinking about that method like a "this method should take highest priority" kind of thing. If you call that method I think it's safe to assume that people probably won't use `setNamespace(String)` after using `usePluginNamespace()` but I added that boolean just in case.
That's why the JavaDocs mention that any call to `CommandAPIConfig#setNamespace(String)` is ignored if `usePluginNamespace` is called.

So, in short: I was thinking of this method as highest priority but if we think that should not be the case, I am happy to remove it.

Reply:
> I don't think the `CommandAPIConfig#usePluginNamespace` field is necessary.

It isn't, but it's a convenience function. We can assume for all intents and purposes that a developer is not going to use `setNamespace()` twice and/or use `usePluginNamespace()` and `setNamespace` together.

It's probably "safer" to simply prevent calling it twice in the first place, but it's an extra safety measure that we could probably ignore. If a developer is complaining about namespaces acting funky, I'm more than happy to tell them to stop writing conflicting code.